pt/lessons/ecto/basics.md

```diff
@@ -1,360 +1,243 @@
-    field(:confirmed, :boolean, default: false)
-    field(:password, :string, virtual: true)
-    field(:password_confirmation, :string, virtual: true)
-
-    timestamps
-  end
-
-  @required_fields ~w(username email password password_confirmation)
-  @optional_fields ~w()
-
-  def changeset(user, params \\ :empty) do
-    user
-    |> cast(params, @required_fields, @optional_fields)
-    |> validate_length(:password, min: 8)
-    |> validate_password_confirmation()
-    |> unique_constraint(:username, name: :email)
-    |> put_change(:encrypted_password, hashpwsalt(params[:password]))
-  end
-
-  defp validate_password_confirmation(changeset) do
-    case get_change(changeset, :password_confirmation) do
-      nil ->
-        password_incorrect_error(changeset)
-
-      confirmation ->
-        password = get_field(changeset, :password)
-        if confirmation == password, do: changeset, else: password_mismatch_error(changeset)
-    end
-  end
-
-  defp password_mismatch_error(changeset) do
-    add_error(changeset, :password_confirmation, "Passwords does not match")
-  end
-
-  defp password_incorrect_error(changeset) do
-    add_error(changeset, :password, "is not valid")
-  end
-end
+iex> person.name
+"Tom"
+iex> Map.get(person, :name)
+"Tom"
+iex> %{name: name} = person
+%Friends.Person{age: 11, name: "Tom"}
+iex> name
+"Tom"
 ```
 
-Melhoramos nossa função `changeset/2` e adicionamos três novas funções auxiliares: `validate_password_confirmation/1`, `password_mismatch_error/1` e `password_incorrect_error/1`.
-
-Como o próprio nome sugere, `changeset/2` cria para nós um novo *changeset*. Nele usamos `cast/3` para converter nossos parâmetros para um *changeset* a partir de um conjunto de campos obrigatórios e opcionais. Então nós validamos a presença dos campos obrigatórios. A seguir validamos o tamanho da senha do *changeset*, a correspondência da confirmação da senha usando a nossa propria função, e a unicidade do nome de usuário. Por último, atualizamos nosso campo `password` no banco de dados. Para tal usamos `put_change/3` para atualizar um valor no *changeset*.
-
-Usar `User.changeset/2` é relativamente simples:
+De maneira similar, podemos atualizar nossos esquemas como poderíamos fazer com
+qualquer outro map ou struct em Elixir:
 
 ```elixir
-alias ExampleApp.{User,Repo}
-
-pw = "passwords should be hard"
-changeset = User.changeset(%User{}, %{username: "doomspork",
-                    email: "sean@seancallan.com",
-                    password: pw,
-                    password_confirmation: pw})
-
-case Repo.insert(changeset) do
-  {:ok, record}       -> # Inserted with success
-  {:error, changeset} -> # Something went wrong
-end
+iex> %{person | age: 18}
+%Friends.Person{age: 18, name: "Tom"}
+iex> Map.put(person, :name, "Jerry"}
+%Friends.Person{age: 11, name: "Jerry"}
 ```
 
-É isso aí! Agora você está pronto para guardar alguns dados.
+Em nossa próxima lição, sobre *changesets*, iremos dar uma olhada em como
```

Comment:
I think we don't need to add italic to "changesets" here, since we are explicitly pointing the reader to the Changeset chapter.